src/iminuit/minimize.py

```diff
@@ -85,7 +85,7 @@ def f(x):
     if bool(jac):
         if jac is True:
             raise ValueError("jac=True is not supported, only jac=callable")
-        assert hasattr(jac, "__call__")
+        assert callable(jac)
```

Comment:
I tried to enable B, and there were a few other things that needed manual fixing, so I left it off this PR. But this change seemed nice.